substrate/frame/asset-conversion/src/lib.rs

```diff
@@ -376,6 +374,186 @@ pub mod pallet {
 			asset2: Box<T::AssetKind>,
 		) -> DispatchResult {
 			let sender = ensure_signed(origin)?;
+			Self::do_create_pool(&sender, *asset1, *asset2)?;
+			Ok(())
+		}
+
+		/// Provide liquidity into the pool of `asset1` and `asset2`.
+		/// NOTE: an optimal amount of asset1 and asset2 will be calculated and
+		/// might be different than the provided `amount1_desired`/`amount2_desired`
+		/// thus you should provide the min amount you're happy to provide.
+		/// Params `amount1_min`/`amount2_min` represent that.
+		/// `mint_to` will be sent the liquidity tokens that represent this share of the pool.
+		///
+		/// NOTE: when encountering an incorrect exchange rate and non-withdrawable pool liquidity,
+		/// batch an atomic call with [`Pallet::add_liquidity`] and
+		/// [`Pallet::swap_exact_tokens_for_tokens`] or [`Pallet::swap_tokens_for_exact_tokens`]
+		/// calls to render the liquidity withdrawable and rectify the exchange rate.
+		///
+		/// Once liquidity is added, someone may successfully call
+		/// [`Pallet::swap_exact_tokens_for_tokens`].
+		#[pallet::call_index(1)]
+		#[pallet::weight(T::WeightInfo::add_liquidity())]
+		pub fn add_liquidity(
+			origin: OriginFor<T>,
+			asset1: Box<T::AssetKind>,
+			asset2: Box<T::AssetKind>,
+			amount1_desired: T::Balance,
+			amount2_desired: T::Balance,
+			amount1_min: T::Balance,
+			amount2_min: T::Balance,
+			mint_to: T::AccountId,
+		) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
+			Self::do_add_liquidity(
+				&sender,
+				*asset1,
+				*asset2,
+				amount1_desired,
+				amount2_desired,
+				amount1_min,
+				amount2_min,
+				&mint_to,
+			)?;
+			Ok(())
+		}
+
+		/// Allows you to remove liquidity by providing the `lp_token_burn` tokens that will be
+		/// burned in the process. With the usage of `amount1_min_receive`/`amount2_min_receive`
+		/// it's possible to control the min amount of returned tokens you're happy with.
+		#[pallet::call_index(2)]
+		#[pallet::weight(T::WeightInfo::remove_liquidity())]
+		pub fn remove_liquidity(
+			origin: OriginFor<T>,
+			asset1: Box<T::AssetKind>,
+			asset2: Box<T::AssetKind>,
+			lp_token_burn: T::Balance,
+			amount1_min_receive: T::Balance,
+			amount2_min_receive: T::Balance,
+			withdraw_to: T::AccountId,
+		) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
+			Self::do_remove_liquidity(
+				&sender,
+				*asset1,
+				*asset2,
+				lp_token_burn,
+				amount1_min_receive,
+				amount2_min_receive,
+				&withdraw_to,
+			)?;
+			Ok(())
+		}
+
+		/// Swap the exact amount of `asset1` into `asset2`.
+		/// `amount_out_min` param allows you to specify the min amount of the `asset2`
+		/// you're happy to receive.
+		///
+		/// [`AssetConversionApi::quote_price_exact_tokens_for_tokens`] runtime call can be called
+		/// for a quote.
+		#[pallet::call_index(3)]
+		#[pallet::weight(T::WeightInfo::swap_exact_tokens_for_tokens(path.len() as u32))]
+		pub fn swap_exact_tokens_for_tokens(
```

Comment:
Why did you decide to change the ordering of these? Makes it hard to see the diff and what changed